project/HmrcBuild.scala

```diff
@@ -47,10 +48,16 @@ object HmrcBuild extends Build {
         Resolver.bintrayRepo("hmrc", "releases"),
         "typesafe-releases" at "http://repo.typesafe.com/typesafe/releases/"
       ),
-      crossScalaVersions := Seq("2.11.7")
+      crossScalaVersions := Seq("2.11.7"),
+      testGrouping in sbt.Test := oneForkedJvmPerTest((definedTests in sbt.Test).value)
     )
     .settings(TwirlKeys.templateImports ++= Seq("play.api.mvc._", "play.api.data._", "play.api.i18n._", "play.api.templates.PlayMagic._"))
     .settings(unmanagedSourceDirectories in sbt.Compile += baseDirectory.value / "src/main/twirl")
+
+  private def oneForkedJvmPerTest(tests: Seq[TestDefinition]) =
```

Comment:
We are wanting to test the same thing if `optimizelyId` exists then the script is rendered otherwise it isn't. The issue here for me is having the Fake Application work added, it feels unnecessary. If you make the call a function by fragmentising it you could test it easily.  Without having to setup a Fake Application and set up config, you just handle things by injecting arguments for the relevant tests. This also means you are testing the one thing you need to test, rather than a file that contains that thing.

Agreed having an arg just for testing is a bit of a smell. Rather than having the `optimizelyId` as a default you could just send it in? This then means you can override it in tests and it feels a little less of a smell just having an argument for testing purposes?

#### optimizelySnippet.scala.html
```
@import uk.gov.hmrc.play.config.OptimizelyConfig._

@(idOpt: Option[String])
  @idOpt.map { id =>
    <script src='@{optimizelyBaseUrl + id}.js' type='text/javascript'></script>
  }
```

#### head.scala.html
```
@import uk.gov.hmrc.play.config.OptimizelyConfig._

@optimizelySnippet(optimizelyProjectId)
```